src/queries/subjects.ts

```diff
@@ -1,13 +1,35 @@
+import { useState, useEffect } from "react";
+
 export interface Subject {
   name: string;
   lastDate: string;
 }
```

Comment:
```suggestion
export interface Subject {
  id: string;
  name: string;
  lastDate: string;
}
```
表示やリクエストに必要な `id` が抜けていたので, ここに追加する必要がありそうです.